File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        #sort each word
        #add them into a dictionary
        #return back the value

        #Time Complexity
        #Sort each word: n * 100 * log(100)
        #Space Complexity: O(n) Dictionary

        anagrams = defaultdict(list)
        for word in strs:
            sortedWord = "".join(sorted(word))
            anagrams[sortedWord].append(word)
        
        return list(anagrams.values())

File: 0049-group-anagrams/0049-group-anagrams.swift
class Solution {
    func groupAnagrams(_ strs: [String]) -> [[String]] {
        var anagrams: [String: [String]] = [:]

        for word in strs {
            let sortedWord = String(word.sorted())
            print(sortedWord)
            if anagrams[sortedWord] == nil {
                anagrams[sortedWord] = []
            } 
            anagrams[sortedWord]!.append(word)
        }

        var results: [[String]] = []
        for (_, anagramList) in anagrams {
            results.append(anagramList)
        }
        return results
    }
}